Sources/PromotedCore/Model/ID.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+
+/**
+ Entity IDs used to set the `*_id` fields in events.
+ Used internally to track metadata. External interfaces should
+ only expose the string values of these IDs.
+ */
+enum ID: Equatable {
```

Comment:
Interesting. I don't think I understand the syntax used in this file. It confused me to see ID as an enum. Looks like it can actually contain the values. I assume this is some swift syntax.

Reply:
Yeah, Swift enums are like a cross between enums in other languages and proto oneofs (or C unions). For example, the Swift enum:
```
enum Foo {
  case bar
  case batman(a: String, b: String)
  case robin(c: Int)
}
```
Is kind of equivalent to:
```
oneof {
  bar: {}
  batman: {
    a: string
    b: string
  }
  robin: {
    c: int
  }
}
```
It's also convenient because enums are pass-by-value, so it's easier to replace them with a struct if we need to.